platooning_strategic/src/platoon_strategic.cpp

```diff
@@ -304,8 +305,31 @@ namespace platoon_strategic
                 break;
             }
 
-            resp.new_plan.maneuvers.push_back(composeManeuverMessage(current_progress, end_dist,  
+            if (cf_lane_change_required_ || leader_lane_change_required_)
+            {
+                ROS_DEBUG_STREAM("Planning a required lane change");
+
+                auto adjacentleft_lanelet = routing_graph->adjacentLeft(current_lanelet);
+                lanelet::ConstLanelet target_lanelet;
+                
+                if (adjacentleft_lanelet)
+                {
+                    target_lanelet = adjacentleft_lanelet.get();
+                    ROS_DEBUG_STREAM("target lanelet id for lane change: " << target_lanelet.id());
+                    auto lc_maneuver = composeLaneChangeManeuverMessage(current_progress, end_dist, speed_progress, target_speed, current_lanelet.id(), target_lanelet.id());
+                    resp.new_plan.maneuvers.push_back(lc_maneuver);
+                }
+                else
+                {  
+                    ROS_DEBUG_STREAM("No adjacent left lanes exist, so no lanechange is planned");
```

Comment:
Or, if this is an unrecoverable or somehow impossible scenario, it might make sense to throw an exception here so that we disengage and alter the operator.

Reply:
changed the log to ROS_WARN. 
It is very unlikely that we run into a situation with no left lane since the road is checked for right/left lanes multiple times throughout the plugin. However, I added some more logic for handling that condition,